tests/label_assertion_tests.py

```diff
@@ -0,0 +1,42 @@
+import unittest
+from test_utils import MockPullRequest
+
+NO_LABELS = None
+MOBILE_PRODUCT_LABELS = ['product/all-users', 'product/custom', 'product/invisible', 'product/internal-or-flagged']
+MULTIPLE_LABELS = ['cross requested', 'product/internal-or-flagged', 'ready for review']
+
+
+# To run (from parent directory): $ python -m unittest tests.LabelAssertionTests
+class LabelAssertionTests(unittest.TestCase):
+
+    def test_pr_with_no_labels_no_requirements(self):
+        pr = MockPullRequest('./tests/resources/no_labels.json')
+        self.assertTrue(pr.validate_labels(NO_LABELS, NO_LABELS, NO_LABELS))
+
+    def test_required_any_passes(self):
+        pr = MockPullRequest('./tests/resources/pr_1857_labels.json')
```

Comment:
nice! 
nit: would have been clearer if you had named the `json` files with clearer names (e.g. `required_any.json`)

Unfortunately this way of referencing a filename isn't compatible with all systems. It would be better to do something like:
`os.path.join(os.path.dirname(__file__), 'tests', 'resources', '{filename}.json')`

Reply:
Re: changing the filenames -- I don't think I agree that that schema would be clearer. There's nothing about the json files themselves that relates to the requirement rules (and in fact I'm using some of them to test multiple rules). Each file is just the labels json for some specific PR, so I think labeling them by their PR # seems correct.